test/Microsoft.AspNetCore.Cors.Test/CorsMiddlewareTests.cs

```diff
@@ -169,14 +169,92 @@ public async Task PreFlight_MatchesPolicy_SetsResponseHeaders()
 
                 // Assert
                 response.EnsureSuccessStatusCode();
-                Assert.Equal(2, response.Headers.Count());
-                Assert.Equal("http://localhost:5001", response.Headers.GetValues(CorsConstants.AccessControlAllowOrigin).FirstOrDefault());
-                Assert.Equal("PUT", response.Headers.GetValues(CorsConstants.AccessControlAllowMethods).FirstOrDefault());
+                Assert.Collection(
+                    response.Headers.OrderBy(h => h.Key),
+                    kvp =>
+                    {
+                        Assert.Equal(CorsConstants.AccessControlAllowHeaders, kvp.Key);
+                        Assert.Equal(new[] { "Header1" }, kvp.Value);
+                    },
+                    kvp =>
+                    {
+                        Assert.Equal(CorsConstants.AccessControlAllowMethods, kvp.Key);
+                        Assert.Equal(new[] { "PUT" }, kvp.Value);
+                    },
+                    kvp =>
+                    {
+                        Assert.Equal(CorsConstants.AccessControlAllowOrigin, kvp.Key);
+                        Assert.Equal(new[] { "http://localhost:5001" }, kvp.Value);
+                    });
             }
         }
 
         [Fact]
-        public async Task PreFlightRequest_DoesNotMatchPolicy_DoesNotSetHeaders()
+        public async Task PreFlight_WithCredentialsAllowed_ReturnsWildcardValues()
+        {
+            // Arrange
+            var policy = new CorsPolicyBuilder("http://localhost:5001")
+                .AllowAnyHeader()
+                .AllowAnyMethod()
+                .AllowCredentials()
+                .Build();
+
+            var hostBuilder = new WebHostBuilder()
+                .Configure(app =>
+                {
+                    app.UseCors("customPolicy");
+                    app.Run(async context =>
+                    {
+                        await context.Response.WriteAsync("Cross origin response");
+                    });
+                })
+                .ConfigureServices(services =>
+                {
+                    services.AddCors(options =>
+                    {
+                        options.AddPolicy("customPolicy", policy);
+                    });
+                });
+
+            using (var server = new TestServer(hostBuilder))
+            {
+                // Act
+                // Preflight request.
+                var response = await server.CreateRequest("/")
+                    .AddHeader(CorsConstants.Origin, "http://localhost:5001")
```

Comment:
Can you avoid using localhost in these tests altogether? It's confusing because some browsers don't do CORS over localhost for http.